extensions/hazelcast-client/deployment/src/main/java/io/quarkus/hazelcast/client/deployment/HazelcastClientProcessor.java

```diff
@@ -0,0 +1,232 @@
+import com.hazelcast.topic.MessageListener;
+
+import io.quarkus.arc.deployment.AdditionalBeanBuildItem;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.ExecutionTime;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.ExtensionSslNativeSupportBuildItem;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.NativeImageResourceBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveHierarchyBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.RuntimeReinitializedClassBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ServiceProviderBuildItem;
+import io.quarkus.deployment.util.ServiceUtil;
+import io.quarkus.hazelcast.client.runtime.HazelcastClientBytecodeRecorder;
+import io.quarkus.hazelcast.client.runtime.HazelcastClientConfig;
+import io.quarkus.hazelcast.client.runtime.HazelcastClientProducer;
+import io.quarkus.jaxb.deployment.JaxbFileRootBuildItem;
+
+/**
+ * @author Grzegorz Piwowarek
+ */
+class HazelcastClientProcessor {
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(FeatureBuildItem.HAZELCAST_CLIENT);
+    }
+
+    @BuildStep
+    void enableSSL(BuildProducer<ExtensionSslNativeSupportBuildItem> ssl) {
+        ssl.produce(new ExtensionSslNativeSupportBuildItem(FeatureBuildItem.HAZELCAST_CLIENT));
+    }
+
+    @BuildStep
+    JaxbFileRootBuildItem enableJaxb() {
+        return new JaxbFileRootBuildItem("com/hazelcast");
+    }
+
+    @BuildStep
+    void registerServiceProviders(BuildProducer<ServiceProviderBuildItem> services) throws IOException {
+        registerServiceProviders(DiscoveryStrategyFactory.class, services);
+        registerServiceProviders(ClientExtension.class, services);
+        registerServiceProviders(JsonFactory.class, services);
+    }
+
+    @BuildStep
+    void setup(BuildProducer<AdditionalBeanBuildItem> additionalBeans) {
+        additionalBeans.produce(AdditionalBeanBuildItem.unremovableOf(HazelcastClientProducer.class));
+    }
+
+    @BuildStep
+    @Record(ExecutionTime.RUNTIME_INIT)
+    HazelcastClientConfiguredBuildItem resolveClientProperties(HazelcastClientBytecodeRecorder recorder,
+            HazelcastClientConfig config) {
+        recorder.configureRuntimeProperties(config);
+        return new HazelcastClientConfiguredBuildItem();
+    }
+
+    @BuildStep
+    void registerConfigurationFiles(BuildProducer<NativeImageResourceBuildItem> resources) {
+        resources.produce(new NativeImageResourceBuildItem(
+                "hazelcast-client.yml",
+                "hazelcast-client-default.xml",
+                "hazelcast-client.yaml",
+                "hazelcast-client.xml"));
```

Comment:
Maybe also watch these files for modification with `HotDeploymentWatchedFileBuildItem`?

Reply:
Good idea, thanks!